### PR TITLE
add platform information

### PR DESCRIPTION
**Pull request by neozwu**
*add logic to distinguish GAE standard java 7 and 8 on both DevServer and Prod
*pick right transport option for a given platform

Verification: deploy GAE java 7 and 8 apps, to both DevServer and PROD

### CODE REVIEW

File: google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud;
+
+public final class PlatformInformation {
```
Comment (garrettjonesgoogle) [inline google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java]: Mark as @InternalApi for now. (We can open it up later if there is demand.)

File: google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  public static final String SPECIFICATION_VERSION =
+      System.getProperty("java.specification.version");
+  public static final String GAE_RUNTIME_VERSION =
+      System.getProperty("com.google.appengine.runtime.version");
+  public static final String RUNTIME_JETTY_LOGGER =
+      System.getProperty("org.eclipse.jetty.util.log.class");
+  public static final String JETTY_LOGGER_ON_GAE8_PROD =
+      "com.google.apphosting.runtime.jetty9.JettyLogger";
+  public static final String JETTY_LOGGER_ON_GAE8_DEVSERVER =
+      " com.google.appengine.development.jetty9.JettyLogger";
```
Comment (garrettjonesgoogle) [inline google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java]: there is a space at the beginning of this string

Comment (neozwu) [inline google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java]: This is unfortunately true. Verified this space in google3.

File: google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java (hunk 1 above)
Comment (neozwu) [inline google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java]: Done

Comment (garrettjonesgoogle) [inline google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java]: I recommend this one

File: google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java (hunk 2 above)
Comment (garrettjonesgoogle) [inline google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java]: Add a comment then - this is very unexpected, and others might think it is a mistake too.

File: google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java (hunk 1 above)
Comment (neozwu) [inline google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java]: done

File: google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java (hunk 2 above)
Comment (neozwu) [inline google-cloud-core/src/main/java/com/google/cloud/PlatformInformation.java]: done

**Comment by garrettjonesgoogle** [comment]
LGTM from the coding perspective, I'll let @lesv or @jabubake speak to the logic being used.

**Comment by lesv** [comment]
Checked both J7 & J8 on Std for both local and deployed.  I didn't check GAE Flex-compat.

**Pull request merged**